task and file of lab5

YuraDnistriansky opened the pull request.

YuraDnistriansky committed:
task for lab_5

Eextreeme commented on lab_5.py:
```diff
@@ -0,0 +1,249 @@
+class Book:
+    """
+    Class 'Book', that has characteristics of a book in a bookshop
+    """
+    def __init__(self, attributes):
+        """
```
Don`t take attributes from dictionary, better make it in parameters. If you don't want pylint had mistake, make pylint exception

YuraDnistriansky replied:
Ок

Eextreeme commented on lab_5.py:
```diff
@@ -0,0 +1,249 @@
+    Class, that has the list of books to sort
+    can add and remove books
+    """
+    def __init__(self):
+        """
+        attributes creation
+        """
+        self.__books = []
```
initialise class Book better, not just list

Eextreeme commented on lab_5.py:
```diff
@@ -0,0 +1,249 @@
+        """
+        returns a string before deleting the object
+        """
+        print(f"You deleted the object, that had element {self.__name}")
```
add new line

YuraDnistriansky replied:
ок

Eextreeme commented on lab_5.py:
```diff
@@ -0,0 +1,249 @@
+    book_shop.add_book(book_2)
+    book_shop.add_book(book_3)
+    book_shop.add_book(book_4)
+    print("_" * 20)
```
do a method better

Eextreeme commented on lab_5.py:
```diff
@@ -0,0 +1,249 @@
+        self.__quantity = attributes.get('quantity')
+        self.__number_of_sales = attributes.get('number_of_sales')
+    @property
+    def name(self):
+        """
+        print name
+        """
+        return self.__name
+    @name.setter
+    def name(self, new_name):
+        """
+        passing new values to name
+        """
+        self.__name = new_name
+    @name.deleter
+    def name(self):
+        """
+        deleting an name
```
better do standard getters, not property, it would better to red

Eextreeme commented on lab_5.py:
```diff
@@ -0,0 +1,249 @@
+        """
+        print("You deleted the object, that was created based on a class BookShop")
+if __name__ == "__main__":
+    book_1_data = {
+        'name': "Atomic Habits",
+        'price': 250,
+        'number_of_pages': 200,
+        'author': "James Clear",
+        'quantity': 30,
+        'number_of_sales': 115
+    }
+    book_1 = Book(attributes=book_1_data)
+    book_2_data = {
+        'name': "My life and work",
+        'price': 320,
+        'number_of_pages': 450,
+        'author': "Henry Ford",
+        'quantity': 50,
+        'number_of_sales': 80
+    }
+    book_2 = Book(attributes=book_2_data)
+    book_3_data = {
+        'name': "The richest man in Babylon",
+        'price': 150,
+        'number_of_pages': 180,
+        'author': "George Clasyson",
+        'quantity': 85,
+        'number_of_sales': 150
+    }
+    book_3 = Book(attributes=book_3_data)
+    book_4_data = {
+        'name': "Influence",
+        'price': 350,
+        'number_of_pages': 520,
+        'author': "Robert Cialdini",
+        'quantity': 25,
+        'number_of_sales': 50
```
Do not write attributes in dictionary

YuraDnistriansky replied:
ok

YuraDnistriansky replied:
ok

OlegNovoselskij commented on lab_5.py:
```diff
@@ -0,0 +1,249 @@
+        """
+        deleting an quantity
+        """
+        del self.__quantity
+    @property
+    def number_of_sales(self):
+        """
+        print number of sales
+        """
+        return self.__number_of_sales
+    @number_of_sales.setter
+    def number_of_sales(self, new_number_of_sales):
+        """
+        passing new values to number of sales
+        """
+        self.__number_of_sales = new_number_of_sales
+    @number_of_sales.deleter
+    def number_of_sales(self):
+        """
+        deleting an number of sales
+        """
+        del self.__number_of_sales
+    def __str__(self):
+        """
+        returns a string representation of the object
+        """
+        return f"{self.__name} by {self.__author} -- Price = {self.__price}" \
+               f" -- Sales = {self.__number_of_sales}"
+    def __repr__(self):
+        """
+        returns representation of the object, that can be used to restore the object
+        """
+        return f"Book({self.__name}, {self.__price}, {self.__number_of_pages}," \
+               f" {self.__author}, {self.__quantity}, {self.__number_of_sales})"
+    def __del__(self):
+        """
+        returns a string before deleting the object
+        """
+        print(f"You deleted the object, that had element {self.__name}")
+class BookShop:
```
un comfortable to read please make an empty line between functions

Eextreeme commented on lab_5.py:
```diff
@@ -0,0 +1,249 @@
+        returns representation of the object, that can be used to restore the object
+        """
+        return f"BookShop ({self.__books})"
+    def __del__(self):
+        """
+        returns a strings befor deleting the object
+        """
+        print("You deleted the object, that was created based on a class BookShop")
```
Not used in code

YuraDnistriansky replied:
ok

YuraDnistriansky replied:
ok

OlegNovoselskij commented on lab_5.py:
```diff
@@ -0,0 +1,249 @@
+        """
+        returns a strings befor deleting the object
+        """
+        print("You deleted the object, that was created based on a class BookShop")
+if __name__ == "__main__":
```
make an empty line between class end if

YuraDnistriansky replied:
ok

OlegNovoselskij reviewed (changes requested):
un comfortable to read please fix it

Eextreeme commented on lab_5.py:
```diff
@@ -0,0 +1,249 @@
+        print price
+        """
+        return self.__price
+    @price.setter
+    def price(self, new_price):
+        """
+        passing new values to price
+        """
+        self.__price = new_price
+    @price.deleter
+    def price(self):
+        """
+        deleting an price
+        """
+        del self.__price
```
It don't uses

YuraDnistriansky replied:
ok

Eextreeme commented on lab_5.py:
```diff
@@ -0,0 +1,249 @@
+        """
+        deleting an name
+        """
+        del self.__name
```
Don't uses in code

Eextreeme commented on lab_5.py:
```diff
@@ -0,0 +1,249 @@
+        print number 0f pages
+        """
+        return self.__number_of_pages
+    @number_of_pages.setter
+    def number_of_pages(self, new_number_of_pages):
+        """
+        passing new values to number of pages
+        """
+        self.__number_of_pages = new_number_of_pages
+    @number_of_pages.deleter
+    def number_of_pages(self):
+        """
+        deleting an number of pages
+        """
+        del self.__number_of_pages
```
It don't uses

YuraDnistriansky replied:
ok

YuraDnistriansky committed:
Update lab_5.py

fixed some bugs pointed out to me in the comments

YuraDnistriansky committed:
Update 2 lab_5.py

For the second time, I fixed the glitches suggested in the comments

YuraDnistriansky committed:
Update 3 lab_5.py

For the third time, I fixed the glitches suggested in the comments. Thrown an error R0913 exception for pylint

zveres-ss commented on lab_5.py:
```diff
@@ -0,0 +1,142 @@
+
+def line():
+    """
+    function is going line
```
extra function to print line into console

zveres-ss commented on lab_5.py:
```diff
@@ -0,0 +1,142 @@
+    # pylint: disable = R0913
+    def __init__(self, name, price, number_of_pages, author, quantity, number_of_sales):
+        """
+        passing values to attributes
```
constructs book

zveres-ss commented on lab_5.py:
```diff
@@ -0,0 +1,142 @@
+
+class BookShop:
+    """
+    Class, that has the list of books to sort
+    can add and remove books
```
represents book shop object
